### PR TITLE
Add Pgpool Complete Docs

### PR DESCRIPTION
**Pull request by MobarakHsn**
Signed-off-by: MobarakHsn <mobarak@appscode.com>

### CODE REVIEW

File: docs/guides/pgpool/reconfigure/overview.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+5. `KubeDB` Ops-manager operator watches the `PgpoolOpsRequest` CR.
+
+6. When it finds a `PgpoolOpsRequest` CR, it pauses the `Pgpool` object which is referred from the `PgpoolOpsRequest`. So, the `KubeDB` Provisioner  operator doesn't perform any operations on the `Pgpool` object during the reconfiguring process.  
+
+7. Then the `KubeDB` Ops-manager operator will replace the existing configuration with the new configuration provided or merge the new configuration with the existing configuration according to the `PgpoolOpsRequest` CR.
+
+8. Then the `KubeDB` Ops-manager operator will restart the related PetSet Pods so that they restart with the new configuration defined in the `PgpoolOpsRequest` CR.
+
+9. After the successful reconfiguring of the `Pgpool`, the `KubeDB` Ops-manager operator resumes the `Pgpool` object so that the `KubeDB` Provisioner  operator resumes its usual operations.
```
Comment (raihankhan) [inline docs/guides/pgpool/reconfigure/overview.md:52]: The lines are broken. Re-check them. Happens in all the overviews.

File: docs/guides/pgpool/scaling/horizontal-scaling/horizontal-ops.md [hunk 2]
```diff
@@ -0,0 +1,438 @@
+---
+title: Horizontal Scaling Pgpool
+menu:
+  docs_{{ .version }}:
+    identifier: pp-horizontal-scaling-ops
+    name: pp-horizontal-scaling-ops
```
Comment (raihankhan) [inline docs/guides/pgpool/scaling/horizontal-scaling/horizontal-ops.md]: Fix the name here - HorizontalScaling OpsRequest

File: docs/guides/pgpool/scaling/vertical-scaling/vertical-ops.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
+---
+title: Vertical Scaling Pgpool
+menu:
+  docs_{{ .version }}:
+    identifier: pp-vertical-scaling-ops
+    name: pp-vertical-scaling-ops
```
Comment (raihankhan) [inline docs/guides/pgpool/scaling/vertical-scaling/vertical-ops.md]: Fix the name here - VerticalScaling OpsRequest

File: docs/guides/pgpool/configuration/using-config-file.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
+---
+title: Run Pgpool with Custom Configuration
+menu:
+  docs_{{ .version }}:
+    identifier: pp-using-config-file-configuration
+    name: Config File
```
Comment (raihankhan) [inline docs/guides/pgpool/configuration/using-config-file.md]: Rename - Customize Configurations

File: docs/guides/pgpool/configuration/using-init-config.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
+---
+title: Run Pgpool with Custom Configuration
+menu:
+  docs_{{ .version }}:
+    identifier: pp-using-init-configuration
+    name: Init Config
```
Comment (raihankhan) [inline docs/guides/pgpool/configuration/using-init-config.md]: Rename - Customize Init Config

File: docs/guides/pgpool/configuration/using-podtemplate.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+  - lifecycle
+
+Read about the fields in details in [PodTemplate concept](/docs/guides/pgpool/concepts/pgpool.md#specpodtemplate),
+
+## Prepare Postgres
+For a Pgpool surely we will need a Postgres server so, prepare a KubeDB Postgres cluster using this [tutorial](/docs/guides/postgres/clustering/streaming_replication.md), or you can use any externally managed postgres but in that case you need to create an [appbinding](/docs/guides/pgpool/concepts/appbinding.md) yourself. In this tutorial we will use 3 node Postgres cluster named `ha-postgres`.
+
+
+## CRD Configuration
+
```
Comment (raihankhan) [inline docs/guides/pgpool/configuration/using-podtemplate.md:70]: Add some more example - 
1. Refer to this blog for adding custom sidecar containers - [custom container](https://appscode.com/blog/post/add-custom-container-to-kubedb-databases/)

2. Using nodeselector
3. Using taint and tolerations

**Review by raihankhan — CHANGES_REQUESTED** [review]

File: docs/guides/pgpool/reconfigure/overview.md (hunk 1 above)
Comment (MobarakHsn) [reply docs/guides/pgpool/reconfigure/overview.md:52]: These seems some internal issues to all the DB docs. Locally in goland all seems fine!

File: docs/guides/pgpool/scaling/horizontal-scaling/horizontal-ops.md (hunk 2 above)
Comment (MobarakHsn) [reply docs/guides/pgpool/scaling/horizontal-scaling/horizontal-ops.md]: Fixed

File: docs/guides/pgpool/scaling/vertical-scaling/vertical-ops.md (hunk 3 above)
Comment (MobarakHsn) [reply docs/guides/pgpool/scaling/vertical-scaling/vertical-ops.md]: Fixed

File: docs/guides/pgpool/configuration/using-config-file.md (hunk 4 above)
Comment (MobarakHsn) [reply docs/guides/pgpool/configuration/using-config-file.md]: Fixed

File: docs/guides/pgpool/configuration/using-init-config.md (hunk 5 above)
Comment (MobarakHsn) [reply docs/guides/pgpool/configuration/using-init-config.md]: Fixed

File: docs/guides/pgpool/configuration/using-podtemplate.md (hunk 6 above)
Comment (MobarakHsn) [reply docs/guides/pgpool/configuration/using-podtemplate.md:70]: Added

**Review by raihankhan — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**